crates/hotshot/src/traits/networking/libp2p_network.rs

```diff
@@ -528,49 +588,36 @@ impl<M: NetworkMsg, K: SignatureKey + 'static> Libp2pNetwork<M, K> {
         let handle = self.clone();
         let is_bootstrapped = self.inner.is_bootstrapped.clone();
         async_spawn(async move {
-            while let Ok(msg) = handle.inner.handle.receiver().recv().await {
-                match msg {
-                    GossipMsg(msg, _topic) => {
-                        let result: Result<M, _> = bincode_opts().deserialize(&msg);
-                        if let Ok(result) = result {
-                            broadcast_send
-                                .send(result)
-                                .await
-                                .map_err(|_| NetworkError::ChannelSend)?;
-                        }
-                    }
-                    DirectRequest(msg, _pid, chan) => {
-                        let result: Result<M, _> = bincode_opts()
-                            .deserialize(&msg)
-                            .context(FailedToSerializeSnafu);
-                        if let Ok(result) = result {
-                            direct_send
-                                .send(result)
-                                .await
-                                .map_err(|_| NetworkError::ChannelSend)?;
-                        }
-                        if handle
-                            .inner
-                            .handle
-                            .direct_response(chan, &Empty::Empty)
-                            .await
-                            .is_err()
-                        {
-                            error!("failed to ack!");
-                        };
-                    }
-                    DirectResponse(msg, _) => {
-                        let _result: Result<M, _> = bincode_opts()
-                            .deserialize(&msg)
-                            .context(FailedToSerializeSnafu);
-                    }
+            while let Ok(message) = handle.inner.handle.receiver().recv().await {
+                match &message {
                     NetworkEvent::IsBootstrapped => {
                         is_bootstrapped.store(true, Ordering::Relaxed);
                     }
+                    GossipMsg(raw, _) | DirectRequest(raw, _, _) | DirectResponse(raw, _) => {
+                        let message_version = read_version(raw);
+                        match message_version {
+                            Some(VERSION_0_1) => {
```

Comment:
Currently, outgoing messages always use `VERSION_0_1` (defined in `constants`) -- I think the current idea is that a single binary will only ever send out messages in a single version.

I think this will come up later in https://github.com/EspressoSystems/HotShot/issues/2444; once we've seen a committed upgrade, the version number on outgoing messages might change (and depending on what we upgrade, we might start sending out new types of messages etc).